R/detect_mistakes.R

```diff
@@ -49,6 +49,7 @@ detect_mistakes <- function(user,
   }
 
   submitted <- user
+  answer <- solution
```

Comment:
Would prefer to not add unneeded variables.

```suggestion
```

Reply:
answer is only needed (at this time) for this instruction : 

```  
explicit_solution <-call_standardise_formals(unpipe_all(answer),env = env,include_defaults = FALSE)
```

I can move up this line to line 52 to avoid variable creation, but it will be less clear than to have this line in step 5 :  

```
  # 5. Check that every named argument in the solution appears in the user code.
  #    The outcome of this order is that when a user writes na = TRUE, gradethis
  #    will tell them that it expected an na.rm argument, not that na is a surplus
  #    argument.
 
  explicit_solution <-call_standardise_formals(unpipe_all(answer),env = env,include_defaults = FALSE)
  explicit_user <-call_standardise_formals(unpipe_all(submitted),env = env,include_defaults = FALSE)
  explicit_user_names <- real_names(explicit_user)
  explicit_solution_names <-  real_names(explicit_solution)
```